6.4.2 fails to build on CPython 3.10.17, with assertions enabled

(the error's on top, included all the warnings for completeness)

```console
$ python3.10 --version
Python 3.10.17
$ uv venv -p 3.10
Using CPython 3.10.17 interpreter at: /usr/bin/python3.10
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate
$ uv pip install --no-binary multidict multidict
Resolved 2 packages in 3ms
  × Failed to build `multidict==6.4.2`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `setuptools.build_meta.build_wheel` failed (exit status: 1)

      [stdout]
      *********************
      * Accelerated build *
      *********************
      running bdist_wheel
      running build
      running build_py
      copying multidict/__init__.py -> build/lib.linux-x86_64-cpython-310/multidict
      copying multidict/_abc.py -> build/lib.linux-x86_64-cpython-310/multidict
      copying multidict/_compat.py -> build/lib.linux-x86_64-cpython-310/multidict
      copying multidict/_multidict_py.py -> build/lib.linux-x86_64-cpython-310/multidict
      copying multidict/py.typed -> build/lib.linux-x86_64-cpython-310/multidict
      running build_ext
      building 'multidict._multidict' extension
      x86_64-pc-linux-gnu-gcc -Wno-unused-result -Wsign-compare -fPIC -I/home/mgorny/.cache/uv/builds-v0/.tmpKxnWII/include
      -I/usr/include/python3.10 -c multidict/_multidict.c -o build/temp.linux-x86_64-cpython-310/multidict/_multidict.o -O3 -std=c99
      -Wall -Wsign-compare -Wconversion -fno-strict-aliasing -pedantic

      [stderr]
      /home/mgorny/.cache/uv/builds-v0/.tmpKxnWII/lib/python3.10/site-packages/setuptools/dist.py:759: SetuptoolsDeprecationWarning:
      License classifiers are deprecated.
      !!

              ********************************************************************************
              Please consider removing the following classifiers in favor of a SPDX license expression:

              License :: OSI Approved :: Apache Software License

              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************

      !!
        self._finalize_license_expression()
      In file included from /usr/include/python3.10/Python.h:48,
                       from multidict/_multidict.c:1:
      multidict/_multilib/state.h: In function ‘PyType_GetModuleByDef’:
      multidict/_multilib/state.h:68:52: error: ‘type’ undeclared (first use in this function); did you mean ‘tp’?
         68 |     assert(PyTuple_GET_ITEM(mro, 0) == (PyObject *)type);
            |                                                    ^~~~
      multidict/_multilib/state.h:68:52: note: each undeclared identifier is reported only once for each function it appears in
      In file included from multidict/_multilib/pair_list.h:15,
                       from multidict/_multilib/dict.h:9,
                       from multidict/_multidict.c:6:
      multidict/_multilib/istr.h: At top level:
      multidict/_multilib/istr.h:98:21: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
         98 |     {Py_tp_dealloc, istr_dealloc},
            |                     ^~~~~~~~~~~~
      multidict/_multilib/istr.h:98:21: note: (near initialization for ‘istr_slots[0].pfunc’)
      multidict/_multilib/istr.h:101:17: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        101 |     {Py_tp_new, istr_new},
            |                 ^~~~~~~~
      multidict/_multilib/istr.h:101:17: note: (near initialization for ‘istr_slots[3].pfunc’)
      In file included from multidict/_multidict.c:8:
      multidict/_multilib/iter.h:184:21: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        184 |     {Py_tp_dealloc, multidict_iter_dealloc},
            |                     ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:184:21: note: (near initialization for ‘multidict_items_iter_slots[0].pfunc’)
      multidict/_multilib/iter.h:186:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        186 |     {Py_tp_traverse, multidict_iter_traverse},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:186:22: note: (near initialization for ‘multidict_items_iter_slots[2].pfunc’)
      multidict/_multilib/iter.h:187:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        187 |     {Py_tp_clear, multidict_iter_clear},
            |                   ^~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:187:19: note: (near initialization for ‘multidict_items_iter_slots[3].pfunc’)
      multidict/_multilib/iter.h:188:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        188 |     {Py_tp_iter, PyObject_SelfIter},
            |                  ^~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:188:18: note: (near initialization for ‘multidict_items_iter_slots[4].pfunc’)
      multidict/_multilib/iter.h:189:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        189 |     {Py_tp_iternext, multidict_items_iter_iternext},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:189:22: note: (near initialization for ‘multidict_items_iter_slots[5].pfunc’)
      multidict/_multilib/iter.h:205:21: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        205 |     {Py_tp_dealloc, multidict_iter_dealloc},
            |                     ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:205:21: note: (near initialization for ‘multidict_values_iter_slots[0].pfunc’)
      multidict/_multilib/iter.h:207:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        207 |     {Py_tp_traverse, multidict_iter_traverse},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:207:22: note: (near initialization for ‘multidict_values_iter_slots[2].pfunc’)
      multidict/_multilib/iter.h:208:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        208 |     {Py_tp_clear, multidict_iter_clear},
            |                   ^~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:208:19: note: (near initialization for ‘multidict_values_iter_slots[3].pfunc’)
      multidict/_multilib/iter.h:209:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        209 |     {Py_tp_iter, PyObject_SelfIter},
            |                  ^~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:209:18: note: (near initialization for ‘multidict_values_iter_slots[4].pfunc’)
      multidict/_multilib/iter.h:210:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        210 |     {Py_tp_iternext, multidict_values_iter_iternext},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:210:22: note: (near initialization for ‘multidict_values_iter_slots[5].pfunc’)
      multidict/_multilib/iter.h:227:21: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        227 |     {Py_tp_dealloc, multidict_iter_dealloc},
            |                     ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:227:21: note: (near initialization for ‘multidict_keys_iter_slots[0].pfunc’)
      multidict/_multilib/iter.h:229:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        229 |     {Py_tp_traverse, multidict_iter_traverse},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:229:22: note: (near initialization for ‘multidict_keys_iter_slots[2].pfunc’)
      multidict/_multilib/iter.h:230:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        230 |     {Py_tp_clear, multidict_iter_clear},
            |                   ^~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:230:19: note: (near initialization for ‘multidict_keys_iter_slots[3].pfunc’)
      multidict/_multilib/iter.h:231:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        231 |     {Py_tp_iter, PyObject_SelfIter},
            |                  ^~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:231:18: note: (near initialization for ‘multidict_keys_iter_slots[4].pfunc’)
      multidict/_multilib/iter.h:232:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        232 |     {Py_tp_iternext, multidict_keys_iter_iternext},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/iter.h:232:22: note: (near initialization for ‘multidict_keys_iter_slots[5].pfunc’)
      In file included from multidict/_multidict.c:12:
      multidict/_multilib/views.h:1010:21: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1010 |     {Py_tp_dealloc, multidict_view_dealloc},
            |                     ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1010:21: note: (near initialization for ‘multidict_itemsview_slots[0].pfunc’)
      multidict/_multilib/views.h:1011:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1011 |     {Py_tp_repr, multidict_itemsview_repr},
            |                  ^~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1011:18: note: (near initialization for ‘multidict_itemsview_slots[1].pfunc’)
      multidict/_multilib/views.h:1013:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1013 |     {Py_nb_subtract, multidict_itemsview_sub},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1013:22: note: (near initialization for ‘multidict_itemsview_slots[2].pfunc’)
      multidict/_multilib/views.h:1014:17: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1014 |     {Py_nb_and, multidict_itemsview_and},
            |                 ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1014:17: note: (near initialization for ‘multidict_itemsview_slots[3].pfunc’)
      multidict/_multilib/views.h:1015:17: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1015 |     {Py_nb_xor, multidict_itemsview_xor},
            |                 ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1015:17: note: (near initialization for ‘multidict_itemsview_slots[4].pfunc’)
      multidict/_multilib/views.h:1016:16: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1016 |     {Py_nb_or, multidict_itemsview_or},
            |                ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1016:16: note: (near initialization for ‘multidict_itemsview_slots[5].pfunc’)
      multidict/_multilib/views.h:1017:20: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1017 |     {Py_sq_length, multidict_view_len},
            |                    ^~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1017:20: note: (near initialization for ‘multidict_itemsview_slots[6].pfunc’)
      multidict/_multilib/views.h:1018:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1018 |     {Py_sq_contains, multidict_itemsview_contains},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1018:22: note: (near initialization for ‘multidict_itemsview_slots[7].pfunc’)
      multidict/_multilib/views.h:1019:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1019 |     {Py_tp_getattro, PyObject_GenericGetAttr},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1019:22: note: (near initialization for ‘multidict_itemsview_slots[8].pfunc’)
      multidict/_multilib/views.h:1020:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1020 |     {Py_tp_traverse, multidict_view_traverse},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1020:22: note: (near initialization for ‘multidict_itemsview_slots[9].pfunc’)
      multidict/_multilib/views.h:1021:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1021 |     {Py_tp_clear, multidict_view_clear},
            |                   ^~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1021:19: note: (near initialization for ‘multidict_itemsview_slots[10].pfunc’)
      multidict/_multilib/views.h:1022:25: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1022 |     {Py_tp_richcompare, multidict_view_richcompare},
            |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1022:25: note: (near initialization for ‘multidict_itemsview_slots[11].pfunc’)
      multidict/_multilib/views.h:1023:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1023 |     {Py_tp_iter, multidict_itemsview_iter},
            |                  ^~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1023:18: note: (near initialization for ‘multidict_itemsview_slots[12].pfunc’)
      multidict/_multilib/views.h:1564:21: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1564 |     {Py_tp_dealloc, multidict_view_dealloc},
            |                     ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1564:21: note: (near initialization for ‘multidict_keysview_slots[0].pfunc’)
      multidict/_multilib/views.h:1565:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1565 |     {Py_tp_repr, multidict_keysview_repr},
            |                  ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1565:18: note: (near initialization for ‘multidict_keysview_slots[1].pfunc’)
      multidict/_multilib/views.h:1567:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1567 |     {Py_nb_subtract, multidict_keysview_sub},
            |                      ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1567:22: note: (near initialization for ‘multidict_keysview_slots[2].pfunc’)
      multidict/_multilib/views.h:1568:17: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1568 |     {Py_nb_and, multidict_keysview_and},
            |                 ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1568:17: note: (near initialization for ‘multidict_keysview_slots[3].pfunc’)
      multidict/_multilib/views.h:1569:17: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1569 |     {Py_nb_xor, multidict_keysview_xor},
            |                 ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1569:17: note: (near initialization for ‘multidict_keysview_slots[4].pfunc’)
      multidict/_multilib/views.h:1570:16: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1570 |     {Py_nb_or, multidict_keysview_or},
            |                ^~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1570:16: note: (near initialization for ‘multidict_keysview_slots[5].pfunc’)
      multidict/_multilib/views.h:1571:20: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1571 |     {Py_sq_length, multidict_view_len},
            |                    ^~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1571:20: note: (near initialization for ‘multidict_keysview_slots[6].pfunc’)
      multidict/_multilib/views.h:1572:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1572 |     {Py_sq_contains, multidict_keysview_contains},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1572:22: note: (near initialization for ‘multidict_keysview_slots[7].pfunc’)
      multidict/_multilib/views.h:1573:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1573 |     {Py_tp_getattro, PyObject_GenericGetAttr},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1573:22: note: (near initialization for ‘multidict_keysview_slots[8].pfunc’)
      multidict/_multilib/views.h:1574:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1574 |     {Py_tp_traverse, multidict_view_traverse},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1574:22: note: (near initialization for ‘multidict_keysview_slots[9].pfunc’)
      multidict/_multilib/views.h:1575:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1575 |     {Py_tp_clear, multidict_view_clear},
            |                   ^~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1575:19: note: (near initialization for ‘multidict_keysview_slots[10].pfunc’)
      multidict/_multilib/views.h:1576:25: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1576 |     {Py_tp_richcompare, multidict_view_richcompare},
            |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1576:25: note: (near initialization for ‘multidict_keysview_slots[11].pfunc’)
      multidict/_multilib/views.h:1577:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1577 |     {Py_tp_iter, multidict_keysview_iter},
            |                  ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1577:18: note: (near initialization for ‘multidict_keysview_slots[12].pfunc’)
      multidict/_multilib/views.h:1638:21: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1638 |     {Py_tp_dealloc, multidict_view_dealloc},
            |                     ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1638:21: note: (near initialization for ‘multidict_valuesview_slots[0].pfunc’)
      multidict/_multilib/views.h:1639:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1639 |     {Py_tp_repr, multidict_valuesview_repr},
            |                  ^~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1639:18: note: (near initialization for ‘multidict_valuesview_slots[1].pfunc’)
      multidict/_multilib/views.h:1641:20: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1641 |     {Py_sq_length, multidict_view_len},
            |                    ^~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1641:20: note: (near initialization for ‘multidict_valuesview_slots[2].pfunc’)
      multidict/_multilib/views.h:1642:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1642 |     {Py_tp_getattro, PyObject_GenericGetAttr},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1642:22: note: (near initialization for ‘multidict_valuesview_slots[3].pfunc’)
      multidict/_multilib/views.h:1643:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1643 |     {Py_tp_traverse, multidict_view_traverse},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1643:22: note: (near initialization for ‘multidict_valuesview_slots[4].pfunc’)
      multidict/_multilib/views.h:1644:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1644 |     {Py_tp_clear, multidict_view_clear},
            |                   ^~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1644:19: note: (near initialization for ‘multidict_valuesview_slots[5].pfunc’)
      multidict/_multilib/views.h:1645:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1645 |     {Py_tp_iter, multidict_valuesview_iter},
            |                  ^~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multilib/views.h:1645:18: note: (near initialization for ‘multidict_valuesview_slots[6].pfunc’)
      multidict/_multidict.c:878:21: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        878 |     {Py_tp_dealloc, multidict_tp_dealloc},
            |                     ^~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:878:21: note: (near initialization for ‘multidict_slots[0].pfunc’)
      multidict/_multidict.c:879:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        879 |     {Py_tp_repr, multidict_repr},
            |                  ^~~~~~~~~~~~~~
      multidict/_multidict.c:879:18: note: (near initialization for ‘multidict_slots[1].pfunc’)
      multidict/_multidict.c:882:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        882 |     {Py_sq_contains, multidict_sq_contains},
            |                      ^~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:882:22: note: (near initialization for ‘multidict_slots[3].pfunc’)
      multidict/_multidict.c:883:20: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        883 |     {Py_mp_length, multidict_mp_len},
            |                    ^~~~~~~~~~~~~~~~
      multidict/_multidict.c:883:20: note: (near initialization for ‘multidict_slots[4].pfunc’)
      multidict/_multidict.c:884:23: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        884 |     {Py_mp_subscript, multidict_mp_subscript},
            |                       ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:884:23: note: (near initialization for ‘multidict_slots[5].pfunc’)
      multidict/_multidict.c:885:27: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        885 |     {Py_mp_ass_subscript, multidict_mp_as_subscript},
            |                           ^~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:885:27: note: (near initialization for ‘multidict_slots[6].pfunc’)
      multidict/_multidict.c:887:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        887 |     {Py_tp_traverse, multidict_tp_traverse},
            |                      ^~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:887:22: note: (near initialization for ‘multidict_slots[7].pfunc’)
      multidict/_multidict.c:888:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        888 |     {Py_tp_clear, multidict_tp_clear},
            |                   ^~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:888:19: note: (near initialization for ‘multidict_slots[8].pfunc’)
      multidict/_multidict.c:889:25: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        889 |     {Py_tp_richcompare, multidict_tp_richcompare},
            |                         ^~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:889:25: note: (near initialization for ‘multidict_slots[9].pfunc’)
      multidict/_multidict.c:890:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        890 |     {Py_tp_iter, multidict_tp_iter},
            |                  ^~~~~~~~~~~~~~~~~
      multidict/_multidict.c:890:18: note: (near initialization for ‘multidict_slots[10].pfunc’)
      multidict/_multidict.c:892:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        892 |     {Py_tp_init, multidict_tp_init},
            |                  ^~~~~~~~~~~~~~~~~
      multidict/_multidict.c:892:18: note: (near initialization for ‘multidict_slots[12].pfunc’)
      multidict/_multidict.c:893:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        893 |     {Py_tp_alloc, PyType_GenericAlloc},
            |                   ^~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:893:19: note: (near initialization for ‘multidict_slots[13].pfunc’)
      multidict/_multidict.c:894:17: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        894 |     {Py_tp_new, PyType_GenericNew},
            |                 ^~~~~~~~~~~~~~~~~
      multidict/_multidict.c:894:17: note: (near initialization for ‘multidict_slots[14].pfunc’)
      multidict/_multidict.c:895:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        895 |     {Py_tp_free, PyObject_GC_Del},
            |                  ^~~~~~~~~~~~~~~
      multidict/_multidict.c:895:18: note: (near initialization for ‘multidict_slots[15].pfunc’)
      multidict/_multidict.c:950:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
        950 |     {Py_tp_init, cimultidict_tp_init},
            |                  ^~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:950:18: note: (near initialization for ‘cimultidict_slots[1].pfunc’)
      multidict/_multidict.c:1208:21: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1208 |     {Py_tp_dealloc, multidict_proxy_tp_dealloc},
            |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1208:21: note: (near initialization for ‘multidict_proxy_slots[0].pfunc’)
      multidict/_multidict.c:1209:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1209 |     {Py_tp_repr, multidict_proxy_repr},
            |                  ^~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1209:18: note: (near initialization for ‘multidict_proxy_slots[1].pfunc’)
      multidict/_multidict.c:1212:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1212 |     {Py_sq_contains, multidict_proxy_sq_contains},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1212:22: note: (near initialization for ‘multidict_proxy_slots[3].pfunc’)
      multidict/_multidict.c:1213:20: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1213 |     {Py_mp_length, multidict_proxy_mp_len},
            |                    ^~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1213:20: note: (near initialization for ‘multidict_proxy_slots[4].pfunc’)
      multidict/_multidict.c:1214:23: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1214 |     {Py_mp_subscript, multidict_proxy_mp_subscript},
            |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1214:23: note: (near initialization for ‘multidict_proxy_slots[5].pfunc’)
      multidict/_multidict.c:1216:22: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1216 |     {Py_tp_traverse, multidict_proxy_tp_traverse},
            |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1216:22: note: (near initialization for ‘multidict_proxy_slots[6].pfunc’)
      multidict/_multidict.c:1217:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1217 |     {Py_tp_clear, multidict_proxy_tp_clear},
            |                   ^~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1217:19: note: (near initialization for ‘multidict_proxy_slots[7].pfunc’)
      multidict/_multidict.c:1218:25: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1218 |     {Py_tp_richcompare, multidict_proxy_tp_richcompare},
            |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1218:25: note: (near initialization for ‘multidict_proxy_slots[8].pfunc’)
      multidict/_multidict.c:1219:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1219 |     {Py_tp_iter, multidict_proxy_tp_iter},
            |                  ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1219:18: note: (near initialization for ‘multidict_proxy_slots[9].pfunc’)
      multidict/_multidict.c:1221:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1221 |     {Py_tp_init, multidict_proxy_tp_init},
            |                  ^~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1221:18: note: (near initialization for ‘multidict_proxy_slots[11].pfunc’)
      multidict/_multidict.c:1222:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1222 |     {Py_tp_alloc, PyType_GenericAlloc},
            |                   ^~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1222:19: note: (near initialization for ‘multidict_proxy_slots[12].pfunc’)
      multidict/_multidict.c:1223:17: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1223 |     {Py_tp_new, PyType_GenericNew},
            |                 ^~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1223:17: note: (near initialization for ‘multidict_proxy_slots[13].pfunc’)
      multidict/_multidict.c:1224:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1224 |     {Py_tp_free, PyObject_GC_Del},
            |                  ^~~~~~~~~~~~~~~
      multidict/_multidict.c:1224:18: note: (near initialization for ‘multidict_proxy_slots[14].pfunc’)
      multidict/_multidict.c:1319:18: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1319 |     {Py_tp_init, cimultidict_proxy_tp_init},
            |                  ^~~~~~~~~~~~~~~~~~~~~~~~~
      multidict/_multidict.c:1319:18: note: (near initialization for ‘cimultidict_proxy_slots[2].pfunc’)
      multidict/_multidict.c:1513:19: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
       1513 |     {Py_mod_exec, module_exec},
            |                   ^~~~~~~~~~~
      multidict/_multidict.c:1513:19: note: (near initialization for ‘module_slots[0].value’)
      error: command '/usr/bin/x86_64-pc-linux-gnu-gcc' failed with exit code 1

      hint: This usually indicates a problem with the package or the build environment.
```

It wasn't immediately obvious when looking on mobile. So here's the copy of the error lines from the snippet above:
```console
      multidict/_multilib/state.h:68:52: error: ‘type’ undeclared (first use in this function); did you mean ‘tp’?
         68 |     assert(PyTuple_GET_ITEM(mro, 0) == (PyObject *)type);
            |                                                    ^~~~
```

--- a/CHANGES/1144.bugfix.rst
+++ b/CHANGES/1144.bugfix.rst
@@ -1,1 +1,1 @@
-Fixed the library building in debug mode.
+Fixed building the library in debug mode.

--- a/CHANGES/1144.bugfix.rst
+++ b/CHANGES/1144.bugfix.rst
@@ -1,0 +1,1 @@
+Fixed the library building in debug mode.

--- a/multidict/_multilib/state.h
+++ b/multidict/_multilib/state.h
@@ -65,7 +65,7 @@ PyType_GetModuleByDef(PyTypeObject *tp, PyModuleDef *def)
     assert(mro != NULL);
     assert(PyTuple_Check(mro));
     assert(PyTuple_GET_SIZE(mro) >= 1);
-    assert(PyTuple_GET_ITEM(mro, 0) == (PyObject *)type);
+    assert(PyTuple_GET_ITEM(mro, 0) == (PyObject *)tp);
 
     Py_ssize_t n = PyTuple_GET_SIZE(mro);
     for (Py_ssize_t i = 1; i < n; i++) {

--- a/setup.py
+++ b/setup.py
@@ -10,7 +10,7 @@ if sys.implementation.name != "cpython":
     NO_EXTENSIONS = True
 
 CFLAGS = ["-O3"]
-# CFLAGS = ["-O0", "-g3", "-DDEBUG"]
+# CFLAGS = ["-O0", "-g3", "-UNDEBUG"]
 if platform.system() != "Windows":
     CFLAGS.extend(
         [

